pkg/metrics/roundtripper.go

```diff
@@ -0,0 +1,151 @@
+package metrics
+
+import (
+	"net/http"
+	"strconv"
+	"strings"
+
+	"github.com/operator-framework/operator-marketplace/pkg/defaults"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+)
+
+// appRegistryRoundTripperCounter wraps a RoundTripperFunc with functions that record prometheus metrics
+// when reaching out to an AppRegistry.
+func appRegistryRoundTripperCounter(counter *prometheus.CounterVec, next http.RoundTripper) promhttp.RoundTripperFunc {
+	return promhttp.RoundTripperFunc(func(r *http.Request) (*http.Response, error) {
+		resp, err := next.RoundTrip(r)
+		if err == nil {
+			if quayNamespace, ok := r.URL.Query()["namespace"]; ok && len(quayNamespace) == 1 {
+				counter.With(labels(r.Method, resp.StatusCode, quayNamespace[0])).Inc()
+			}
+		}
+		return resp, err
+	})
+}
+
+// labels adds the expected labels for the AppRegistry Counter.
+func labels(reqMethod string, status int, quayNamespace string) prometheus.Labels {
+	labels := prometheus.Labels{}
+	labels["code"] = sanitizeCode(status)
+	labels["method"] = strings.ToLower(reqMethod)
+	labels["opsrc"] = getOpsrcLabelValue(quayNamespace)
+
+	return labels
+}
+
+// getOpsrcLabelValue returns the quayNamespace value if it
+// matches a default OperatorSource. Otherwise, a generic
+// name is returned to prevent leaking customer data to telemetry.
+func getOpsrcLabelValue(quayNamespace string) string {
+	if defaults.IsDefaultSource(quayNamespace) {
```

Comment:
Do we need to do this? Couldn't we report everything and then in whatever alert we set up drill down into the defaults we want? Or do we want to limit the data we are actually recording?

Reply:
This is to limit the data that we are recording.